app_dart/lib/src/model/appengine/branch.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:cocoon_service/src/model/appengine/key_converter.dart';
+import 'package:gcloud/db.dart';
+import 'package:github/github.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'branch.g.dart';
+
+@Kind(name: 'Branch', idType: IdType.String)
+class Branch extends Model<String> {
+  Branch({Key<String>? key, required int lastActivity}) {
+    parentKey = key?.parent;
+    id = key?.id;
+  }
+
+  /// The timestamp (in milliseconds since the Epoch) of the last time
+  /// when current branch had activity.
+  @IntProperty(propertyName: 'lastActivity', required: true)
+  int? lastActivity;
+
+  /// [RepositorySlug] of where this commit exists.
+  RepositorySlug get slug => RepositorySlug.full(repository);
+
+  String get repository => key.id!.substring(0, key.id!.lastIndexOf('/'));
+
+  String get branch => key.id!.substring(key.id!.lastIndexOf('/') + 1);
```

Comment:
Can we add a channel field as well? In the future, we can use this to get the current candidate branch for CPs

Reply:
Sure, added. I wasn't very sure about the use of channel. I thought if we have a channel name such as beta, then it would already manifest itself as being 'beta' in the branch name field? umm I am just wondering what's the difference between branch name and channel